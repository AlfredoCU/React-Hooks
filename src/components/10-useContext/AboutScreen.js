import React, { useContext } from "react";
import { UserContext } from "./UserContext";

const AboutScreen = () => {
  const { user, setUser } = useContext(UserContext);
  console.log(user);

  const handleLogout = () => {
    setUser({});
  };

  return (
    <div>
      <p>AboutScreen</p>
      <p>{JSON.stringify(user, null, 3)}</p>
      <button onClick={handleLogout} className="btn btn-warning">
        Cerrar sesi√≥n
      </button>
    </div>
  );
};
export default AboutScreen;
