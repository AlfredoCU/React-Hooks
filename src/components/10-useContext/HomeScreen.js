import React, { useContext } from "react";
import { UserContext } from "./UserContext";

const HomeScreen = () => {
  const { user } = useContext(UserContext);
  console.log(user);

  return (
    <div>
      <p>HomeScreen</p>
      <p>{JSON.stringify(user, null, 3)}</p>
    </div>
  );
};

export default HomeScreen;
