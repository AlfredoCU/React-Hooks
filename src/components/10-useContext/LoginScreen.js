import React, { useContext } from "react";
import { UserContext } from "./UserContext";

const LoginScreen = () => {
  const { user, setUser } = useContext(UserContext);

  const handleUser = () => {
    setUser({ ...user, id: 7557, name: "Alfredo", email: "alfred@gmail.com" });
  };
  return (
    <div>
      <h1>LoginScreen</h1>
      <button onClick={handleUser} className="btn btn-success">
        Iniciar Sesi√≥n
      </button>
    </div>
  );
};

export default LoginScreen;
