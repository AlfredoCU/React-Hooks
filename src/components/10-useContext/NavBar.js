import React from "react";
import { NavLink } from "react-router-dom"; // Link

const NavBar = () => {
  return (
    <nav className="navbar navbar-expand-lg navbar-light bg-light">
      <NavLink exact to="/" className="navbar-brand">
        Navbar
      </NavLink>
      <button
        className="navbar-toggler"
        type="button"
        data-toggle="collapse"
        data-target="#navbarNavAltMarkup"
        aria-controls="navbarNavAltMarkup"
        aria-expanded="false"
        aria-label="Toggle navigation"
      >
        <span className="navbar-toggler-icon"></span>
      </button>
      <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
        <div className="navbar-nav">
          <NavLink
            exact
            activeClassName="active"
            to="/"
            className="nav-item nav-link"
          >
            Home
          </NavLink>
          <NavLink
            exact
            activeClassName="active"
            to="/about"
            className="nav-item nav-link"
          >
            About
          </NavLink>
          <NavLink
            exact
            activeClassName="active"
            to="/login"
            className="nav-item nav-link"
          >
            Login
          </NavLink>
        </div>
      </div>
    </nav>
  );
};

export default NavBar;
