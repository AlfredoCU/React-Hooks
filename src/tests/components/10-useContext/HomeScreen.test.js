import React from "react";
import { mount } from "enzyme";
import "@testing-library/jest-dom";
import HomeScreen from "../../../components/10-useContext/HomeScreen";
import { UserContext } from "../../../components/10-useContext/UserContext";

describe("Pruebas en HomeScreen", () => {
  const user = {
    id: 7557,
    name: "Alfredo",
    email: "alfred@gmail.com",
  };

  const wrapper = mount(
    <UserContext.Provider value={{ user }}>
      <HomeScreen />
    </UserContext.Provider>
  );

  test("Foto del componente", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
