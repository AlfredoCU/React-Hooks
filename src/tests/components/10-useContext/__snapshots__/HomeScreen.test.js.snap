// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en HomeScreen Foto del componente 1`] = `
<div>
  <p>
    HomeScreen
  </p>
  <p>
    {
   "name": "Alfred",
   "email": "alf@gmail.com"
}
  </p>
</div>
`;
